lib/sender.js

```diff
@@ -1027,6 +1027,16 @@ class Sender extends EventEmitter {
                         (Date.now() - startTime) / 1000
                     );
                     connection.connected = false;
+
+                    // Give some time to clear connection
+                    // There a some buggy clients with a buggy TCP stack.
+                    // We are destroying the socket to remove FIN_WAIT2 connections and freeing up FDs and memory
+                    setTimeout(() => {
```

Comment:
At this point should the connection be already closed normally? Maybe set timeout only if the socket is still in a waiting state and not for every connection?